wkcuber/api/properties.py

```diff
@@ -0,0 +1,220 @@
+from os.path import join, isfile
+from pathlib import Path
+from typing import Dict, Tuple, List, Optional, Union, Any, Callable
+
+import attr
+import cattr
+import numpy as np
+import wkw
+from cattr.gen import make_dict_unstructure_fn, override, make_dict_structure_fn
+
+from wkcuber.mag import Mag
+from wkcuber.api import BoundingBox
+
+
+def _extract_num_channels(
+    num_channels_in_properties: Optional[int],
+    path: Path,
+    layer: str,
+    mag: Optional[Union[int, Mag]],
+) -> int:
+    # if a wk dataset is not created with this API, then it most likely doesn't have the attribute 'num_channels' in the
```

Comment:
Please use the new `numChannels` name consistently when referring to the json key:

```suggestion
    # if a wk dataset is not created with this API, then it most likely doesn't have the attribute 'numChannels' in the
```